src/styles/main.scss

```diff
@@ -1,3 +1,62 @@
+@import './variables';
+
+html {
+  font-family: Arial, Helvetica, sans-serif;
+}
+
 body {
+  display: flex;
+  justify-content: center;
+  align-items: center;
   margin: 0;
+  height: 100vh;
+}
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $day-cell-gap;
+  width: $row-width;
+  padding: 10px;
+
+  &__day {
+    position: relative;
+    display: none;
+    box-sizing: border-box;
+    min-width: $day-cell-size;
+    height: $day-cell-size;
+    border: 1px solid black;
+    background-color: #eee;
+
+    transition: all 0.5s;
```

Comment:
Don't set transition to all properties, specify to which properties it should apple, in your case `transform` and `background-color`